src/formatDate.js

```diff
@@ -8,7 +8,45 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateArr = date.split(fromFormat[3]);
+  const objToFormat = {};
+  const objFromFormat = {};
+  const newFormat = [];
+  const getLastTwoDigits = (num) => {
+    return num % 100;
+  };
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    objToFormat[toFormat[i]] = i;
+  }
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    objFromFormat[fromFormat[i]] = i;
+  }
+
+  for (const [toKey, toValue] of Object.entries(objToFormat)) {
+    for (const [fromKey, fromValue] of Object.entries(objFromFormat)) {
+      switch (true) {
+        case fromKey === toKey:
+          newFormat[toValue] = dateArr[fromValue];
+          break;
+
+        case toKey === 'YY' && fromKey === 'YYYY':
+          newFormat[toValue] = getLastTwoDigits(+dateArr[fromValue]);
```

Comment:
When converting the year from `YYYY` to `YY`, the current implementation will lead to incorrect formatting for years that are less than 1000, as it will not preserve leading zeros. Use `String.prototype.slice` to extract the last two digits instead.